pkg/controller/core/workload_controller.go

```diff
@@ -235,20 +241,11 @@ func (r *WorkloadReconciler) SetupWithManager(mgr ctrl.Manager) error {
 }
 
 func workloadStatus(w *kueue.Workload) string {
-	if workloadFinished(w) {
+	if workload.WorkloadInCondition(w, kueue.WorkloadFinished) {
 		return finished
 	}
 	if w.Spec.Admission != nil {
 		return admitted
 	}
 	return pending
 }
-
-func workloadFinished(w *kueue.Workload) bool {
-	for _, c := range w.Status.Conditions {
-		if c.Type == kueue.WorkloadFinished {
-			return c.Status == corev1.ConditionTrue
-		}
-	}
```

Comment:
I'd like to keep the function `WorkloadInCondition` but change the implement as you advised to reduce redundant. WDYT?

Reply:
sg, but please rename it something like `InCondition`, similarly we can rename `UpdateWorkloadStatusIfChanged` to `UpdateStatusIfChanged` since they are in the workload pkg already.

Reply:
thanks, all addressed.